dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/JdbcEventStore.java

```diff
@@ -911,15 +880,17 @@ private StringBuilder getFromWhereClause(
         .append("left join trackedentity te on te.trackedentityid=en.trackedentityid ")
         .append("left join userinfo au on (ev.assigneduserid=au.userinfoid) ");
 
-    // JOIN attributes we need to filter on.
-    fromBuilder.append(joinAttributeValue(params, mapSqlParameterSource));
-
-    // LEFT JOIN not filterable attributes we need to sort on.
-    fromBuilder.append(getFromSubQueryJoinOrderByAttributes(params));
+    // LEFT JOIN attributes we need to sort on and/or filter by.
+    fromBuilder.append(getLeftJoinFromAttributes(params));
 
     fromBuilder.append(getCategoryOptionComboQuery(user));
 
-    fromBuilder.append(dataElementAndFiltersSql);
+    fromBuilder.append(dataElementFiltersSql);
+
+    if (!params.getAttributes().isEmpty()) {
```

Comment:
Do we need this conditional? `getLeftJoinFromAttributes` and all the other methods before it do not need it. I would hope the `getWhereClauseFromAttributeFilterConditions` can already deal with an empty `params.getAttributes()`.